### PR TITLE
build(server): set up prometheus metrics

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
Set up Prometheus

/deploy #persist

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
- Move the setup to a separate function specific for setting up monitoring.
- Activate `includePath` and set it up https://www.npmjs.com/package/express-prom-bundle
- We need to setup the annotations on the pod. E.G: this is the current one, ofc to adapt
```
metadata:
  annotations:
    prometheus.io/path: /ui-server/metrics
    prometheus.io/port: '8080'
    prometheus.io/scrape: 'true'
```

File: server/src/config.ts [hunk 1]
```diff
@@ -99,10 +99,16 @@ const WEBSOCKET = {
   timeoutActivationStatus: 120, // in minutes
 };
 
+const PROMETHEUS = {
+  enabled: [true, "true"].includes(process.env.PROMETHEUS_ENABLED),
```
Comment (ciyer) [inline server/src/config.ts:103]: I believe process.env values will always be strings, so no need to check for the boolean, but maybe lowercase and/or allow 1 or "yes". 🤷

File: server/src/utils/prometheus/prometheus.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+import express from "express";
+import promBundle from "express-prom-bundle";
+
+import config from "../../config";
+import logger from "../../logger";
+
+export function initializePrometheus(app: express.Application): void {
+  if (config.prometheus.enabled === false) {
```
Comment (ciyer) [inline server/src/utils/prometheus/prometheus.ts:26]: `config.prometheus` is a boolean, so I think this is simpler. 
```suggestion
  if (!config.prometheus.enabled) {
```

**Review by ciyer — CHANGES_REQUESTED** [review]
It will be nice to be able to access Prometheus metrics! I have some minor code suggestions.

File: server/src/config.ts (hunk 1 above)
Comment (lorenzo-cavazzi) [reply server/src/config.ts:103]: Fair point, I guess "1" is also pretty common. What about using this?

```
enabled: ["true", "1"].includes(
    (process.env.PROMETHEUS_ENABLED ?? "").toLowerCase()
),
```

**Review by ciyer — APPROVED** [review]
LGTM! 🎉

**Pull request merged**